teacher/views.py

```diff
@@ -1,9 +1,80 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
 from home.decorators import group_required
-from home.models.models import Announcement
+from home.models.models import Announcement, Form, Notification, Classroom, Session, Resource
+from home.forms import AddResourceForm
+from attendance.views import get_date_from_template_returned_string
+from staff.staff_views_helper import mark_announcement_dismissed, remove_dismissed_announcements, \
+    remove_submitted_forms, mark_notification_acknowledged
+import os
+from django.http import HttpResponse, Http404
+from django.contrib import messages
+from django.contrib.auth.models import Group
 
 
 @group_required("teacher")
 def teacher(request):
-    announcements = Announcement.objects.filter(recipient_groups=2)
-    return render(request, "teacher.html", {"announcements": announcements})
+    if request.method == "POST":
+        if request.POST.get("dismiss_announcement") == "true":
+            mark_announcement_dismissed(Announcement.objects.get(id=request.POST.get("announcement")), request.user)
+            return redirect("teacher")
+        elif request.POST.get("acknowledge_notification") == "true":
+            mark_notification_acknowledged(Notification.objects.get(id=request.POST.get("notification")))
+            return redirect("teacher")
+    announcements = Announcement.objects.filter(recipient_groups=Group.objects.get(name="teacher").id)
+    announcements = remove_dismissed_announcements(announcements, request.user)
+    forms = Form.objects.filter(recipient_groups=Group.objects.get(name="teacher").id)
+    forms = remove_submitted_forms(forms, request.user)
+    notifications = Notification.objects.filter(user_id=request.user.id)
+    return render(request, "teacher.html", {"announcements": announcements,
+                                            "forms": forms,
+                                            "notifications": notifications})
+
+
+@group_required("teacher")
+def download_form(request):
+    path = request.GET.get("path")
+    file_path = os.path.join(path)
+    if os.path.exists(file_path):
+        with open(file_path, 'rb') as fh:
+            response = HttpResponse(fh.read(), content_type="pdf/text")
+            response['Content-Disposition'] = 'inline; filename=' + os.path.basename(file_path)
+            return response
+    raise Http404
+
+
+@group_required("teacher")
+def my_class(request):
+    try:
+        classroom = Classroom.objects.get(teacher_id=request.user.id)
+    except Exception:
```

Comment:
`except Exception` should be avoided at all times as it's very good at hiding bugs.

This kind of query can be done in one shot using `|` between two query sets or `Q`: https://books.agiliq.com/projects/django-orm-cookbook/en/latest/or_query.html

For the same semantics as this block you'd use `result[:1].get()` or `result[0]` on the result of that query to get the first matching classroom or raise an exception if none match.

Something like this:

```python
classroom = (
    Classroom.objects.filter(teacher_id=request.user.id) |
    Classroom.objects.filter(teacher_assistant_id=request.user.id)
)[0]
```

Although if the model is refactored to unify the way people are associated with classrooms this would be even easier to do.